### PR TITLE
feat(drivers): implement Consensus Research agent driver

### LINKED ISSUE #68: driver: consensus (Research)
## Summary

Implement a driver for the **Consensus** Research agent.

## Agent profile

- **Domain:** Research | Rank 5 | Rating 3.5
- **Capability:** Synthesizes answers from 200M+ academic papers
- **API status:** **No public API** as of 2026. Consumer web app + ChatGPT plugin only.

## Alternative: Semantic Scholar API

Since Consensus has no public API, implement against **Semantic Scholar** (`api.semanticscholar.org`), which provides the same core capability. See #89 for the overall strategy on agents without public APIs.

### Semantic Scholar API

- **Base URL:** `https://api.semanticscholar.org/graph/v1`
- **Auth:** Optional `RUNE_CONSENSUS_API_KEY` (Semantic Scholar API key). Free tier: 100 req/sec.
- **Endpoints:**
  - `GET /paper/search?query=...&limit=10&fields=title,abstract,year,authors,citationCount,url`
  - `GET /paper/{paperId}?fields=...` — full paper details
  - `GET /paper/{paperId}/citations` — citation graph
- **Input:** Query string
- **Output:** JSON with structured paper metadata

### `rune_bench/drivers/consensus/__main__.py`

```
Supported actions:
  ask:
    params: question (str), model (str), ollama_url (str, optional), limit (int, default 10)
    result: {"answer": str, "papers": list[dict], "consensus_score": float | null}
  info:
    result: {"name": "consensus", "version": "1", "actions": ["ask", "info"]}
```

Implementation:
1. Search Semantic Scholar for relevant papers using the `question`
2. Retrieve top N paper abstracts
3. **If `model` + `ollama_url` provided:** Use Ollama to synthesize a consensus answer from the abstracts (mimicking Consensus app's synthesis feature)
4. **If no LLM:** Return raw paper abstracts concatenated
5. Compute a simple "consensus score" based on citation-weighted agreement (optional)

### Synthesis prompt (for Ollama)

```
You are a research synthesis agent. Given these paper abstracts, provide:
1. A clear answer to the research question
2. The level of scientific consensus (strong/moderate/weak/conflicting)
3. Key citations supporting each position

Question: {question}

Papers:
{formatted_abstracts}
```

## Core dependencies

- #84, #85, #88, #89 (no-public-API strategy)
- This is a **hybrid agent**: uses Semantic Scholar for data + optionally Ollama for synthesis

## Agent-specific considerations

- Only agent where we substitute an alternative service for one that has no API
- `model` and `ollama_url` ARE used — for synthesis step (optional but recommended)
- Without Ollama, the driver still works but returns raw paper data only
- Semantic Scholar is free and doesn't require authentication for basic use

## Test plan

- Unit test: mock Semantic Scholar API, verify search + paper parsing
- Unit test: mock Ollama synthesis, verify prompt construction
- Unit test: verify fallback behavior when no LLM configured
- Integration test: end-to-end with real Semantic Scholar API (no key needed)

Part of epic #55.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by lpasquali**
## Summary
- Implement the Consensus Research agent driver using Semantic Scholar API for paper search and optional Ollama-based answer synthesis
- Create `ConsensusDriverClient` (client) and `__main__.py` (subprocess) following the Holmes driver wire protocol pattern
- Replace the `ConsensusRunner` stub with an alias to `ConsensusDriverClient`
- Add 16 tests covering search, paper parsing, Ollama synthesis, fallback formatting, empty results, client transport delegation, and the stdin/stdout main loop

Closes #68

## Test plan
- [x] `python -m pytest tests/test_consensus_driver.py -v --no-cov` — 16/16 passing
- [ ] Manual integration test with live Semantic Scholar API
- [ ] Manual integration test with Ollama synthesis endpoint

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: rune_bench/drivers/pagerduty/__main__.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    url = f"{_PAGERDUTY_API_BASE}{path}"
+    req = urllib.request.Request(
+        url,
+        headers={
+            "Authorization": f"Token token={api_key}",
+            "Content-Type": "application/json",
+        },
+    )
+    with urllib.request.urlopen(req) as resp:  # noqa: S310
+        return json.loads(resp.read().decode())
```
[the remark this answers is not in the document]
Comment (lpasquali) [reply rune_bench/drivers/pagerduty/__main__.py]: Fixed in daea0bc — added `timeout=30` to `_pd_request` urlopen call, consistent with other drivers' API call timeouts.
